src/audio/data_blob.c

```diff
@@ -22,17 +22,40 @@ struct comp_data_blob_handler {
 	uint32_t data_pos;	/**< indicates a data position in data
 				  *  sending/receiving process
 				  */
+	uint32_t single_blob:1; /**< Allocate only one blob. Module can not
+				  *  be active while reconfguring.
```

Comment:
im curious, why can it not be active?

Reply:
The whole feature is for components with very big configurations. In those cases we can not afford to have two copies of the configuration blob in the DSP memory. This also means that when we receive a new configuration - in multiple IPC fragments - we must release the old configuration first (or reuse the old buffer). This means that we do not have a valid configuration during the update so the audio pipeline must be inactive.

Reply:
Ack, can we get that clarified in either comments or some docs?

Reply:
There is already something in comp_data_blob_handler_new_ext() doxygen documentation, but the motivation for the feature is missing. I'll add one more sentence about saving DSP memory.

Reply:
@MingJenTai FYI, so you don't need to do it in your own blob. This is also safer as we can fuzz this